src/components/ui-elements/Navbar.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import Button from "./Button";
+
+const Navbar = () => {
+  const [isDarkMode, setIsDarkMode] = useState(() => {
+    const savedMode = localStorage.getItem("isDarkMode");
+    return savedMode === "true" || false;
+  });
+
+  useEffect(() => {
+    document.body.classList.toggle("dark-theme", isDarkMode);
```

Comment:
We don't access the DOM like we used to in vanilla. 
We use a condition regarding the state or with useRef in extreme cases.